packages/wallet/src/TxBuilder/buildTx.ts

```diff
@@ -189,6 +199,46 @@ export class ObservableWalletTxBuilder implements TxBuilder {
     return [];
   }
 
+  async #addDelegationCertificates(): Promise<void> {
+    if (!this.#delegateConfig) {
+      // Delegation was not configured by user
+      return Promise.resolve();
+    }
+
+    const rewardAccounts = await firstValueFrom(this.#observableWallet.delegation.rewardAccounts$);
+
+    if (!rewardAccounts?.length) {
+      // This shouldn't happen
+      throw new IncompatibleWalletError();
+    }
+
+    // Discard previous delegation and prepare for new one
+    this.partialTxBody = { ...this.partialTxBody, certificates: [] };
+
+    for (const rewardAccount of rewardAccounts) {
+      const stakeKeyHash = Cardano.Ed25519KeyHash.fromRewardAccount(rewardAccount.address);
+      if (this.#delegateConfig.type === 'deregister' && rewardAccount.keyStatus !== StakeKeyStatus.Unregistered) {
```

Comment:
Would be great to `logger.warn` here if `rewardAccount.keyStatus === StakeKeyStatus.Unregistered`

Reply:
[8e14559](https://github.com/input-output-hk/cardano-js-sdk/pull/427/commits/8e145591f64f1a8596d23642196267409543cae3)